docs/candidate-app/password-validation.md

```diff
@@ -0,0 +1,101 @@
+More details on how the different requirements are shown to the user can be found in the documentation of the component.
+
+The UI validation is done using debounced validation, i.e. the validation status is checked and updated once the user stops typing for a moment.
+
+The password validation UI provides a variable indicating the password validation status, which can be used to enable/disable the password submit button. This way frontend allows the password to be submitted only if it is valid.
+
+However, due to the use of debounced validation, the validity of the password is also checked by the password set page itself before sending the request to the backend.
+
+If the validity check passes, a POST request is sent to the backend to either
+
+- `api/auth/candidate/register`
+- `api/auth/reset-password`
+
+## Backend
+
+Before accepting the password, the backend also validates the password.
+This is done in the [`candidate.ts`](/backend/vaa-strapi/src/extensions/users-permissions/controllers/candidate.ts) file using the same `validatePassword` function from [`passwordValidationCopy.ts`](/backend/vaa-strapi/src/util/passwordValidationCopy.ts).
+Currently, the backend uses a copy of the original validation file.
+
+## Password requirements
+
+The password requirements are defined in the [`passwordValidation.ts`](/frontend/src/lib/utils/passwordValidation.ts) file.
+
+Each requirement is defined by the `ValidationDetail` interface:
+
+```ts
+export interface ValidationDetail {
+  status: boolean;
+  message: string;
+  negative?: boolean;
+  enforced?: boolean;
+}
+```
+
+- status:
+  - indicates the state of the requirement, i.e. if it is valid
+- message:
+  - describes the requirement
+  - shown to the user in the validation UI
+  - due to localization, the message string contains the key to the `i18n` localization file, where the actual texts are located
+- negative:
+  - requirements are either positive or negative
+  - positive rules are the main requirements that are always enforced and must be met for the password to be valid
+  - negative rules are rules that are used to prevent bad password practises
+  - in the validation UI, negative rules are shown if they are violated
+- enforced:
+  - negative rules can be either enforced or non-enforced
+  - enforced requirements need to be valid for a password to be valid
+
+Currently, all requirements are done without hardcoded lists of allowed characters and support languages that have separate uppercase and lowercase letters. Localization to other languages may need new validation rules.
+
+The file also contains help functions for checking different aspects of the password string.
+
+### Changing and creating new requirements
+
+All validation rules are defined in the `passwordValidation` function. New requirements can be added and existing ones changed by modifying `ValidationDetail` objects of the result.
```

Comment:
Possibly may want to emphasize that both passwordValidation.ts and passwordValidationCopy.ts need to be updated until both of the files are merged to prevent accidents of doing it only on the frontend (and user may scroll directly to this heading and skip the previous explanation about this)